File: DecompiledGameScripts/cyberpunk/UI/tooltips/cyberwareTooltipData.swift

public class CyberwareTooltipData extends ATooltipData {

  public let label: String;

  public let slotData: [ref<CyberwareSlotTooltipData>];

  public final func AddCyberwareSlotItemData(const itemData: script_ref<InventoryItemData>) -> Void {
    let data: ref<CyberwareSlotTooltipData> = new CyberwareSlotTooltipData();
    data.Empty = InventoryItemData.IsEmpty(itemData);
    if !data.Empty {
      data.Name = InventoryItemData.GetName(itemData);
      data.Description = InventoryItemData.GetDescription(itemData);
      data.IconPath = InventoryItemData.GetIconPath(itemData);
    };
    ArrayPush(this.slotData, data);
  }
}

File: Definitions/Game_Definitions/classes/CyberwareTooltipData.lua
---@meta
---@diagnostic disable

---@class CyberwareTooltipData : ATooltipData
---@field label String
---@field slotData CyberwareSlotTooltipData[]
CyberwareTooltipData = {}

---@return CyberwareTooltipData
function CyberwareTooltipData.new() return end

---@param props table
---@return CyberwareTooltipData
function CyberwareTooltipData.new(props) return end

---@param itemData gameInventoryItemData
function CyberwareTooltipData:AddCyberwareSlotItemData(itemData) return end

